common/rpc/encryption/localStoreTlsProvider.go

```diff
@@ -238,32 +239,45 @@ func newServerTLSConfig(
 	certProvider CertProvider,
 	perHostCertProviderMap PerHostCertProviderMap,
 	config *config.GroupTLS,
+	logger log.Logger,
 ) (*tls.Config, error) {
 
 	clientAuthRequired := config.Server.RequireClientAuth
-	tlsConfig, err := getServerTLSConfigFromCertProvider(certProvider, clientAuthRequired)
+	tlsConfig, err := getServerTLSConfigFromCertProvider(certProvider, clientAuthRequired, "", logger)
```

Comment:
this call feels like it will cause pollution of our logs as this function will log the same thing in the GetConfigForClient function right? Maybe we should not log on this line?

Reply:
This one will call once - when the tls.Config is returned to the gRPC stack as part of configuring the handler.
The func that is set to GetConfigForClient on the other hand gets called upon every connection attempt.
I guess I could condition logging on empty string remote address to not log the first time.